### DISCUSSION TITLE
Only format modified text

### DISCUSSION THREAD

**Discussion by shahamran** [helix-editor/helix#8077] [Ideas]
Add an option to restrict code formatting only to modified lines, as reported by source control.

Quoting [VSCode v1.49 release notes (August 2020)][1]:

> When contributing to another project, the Format on Save and Format Document commands can cause unwanted updates because they format lines that you have not changed. This behavior can result in large pull requests, which are hard to review, and touch code you didn't intend to modify. There is now a new feature to help limit unwanted changes:
> * We added a new command Format Modified Lines, which allows you to only format new and changed lines.
> * There is a new setting editor.formatOnSaveMode, which makes Format on Save only apply to modified lines.

[1]: https://code.visualstudio.com/updates/v1_49#_only-format-modified-text

**Comment by kirawi** [comment]
This would probably require https://github.com/helix-editor/helix/issues/3209
